Modules/xSharePoint/Modules/xSharePoint.Util/xSharePoint.Util.psm1

```diff
@@ -26,6 +26,10 @@ function Get-xSharePointAssemblyVersion() {
     )
     return (Get-Command $PathToAssembly).FileVersionInfo.FileMajorPart
 }
+function Get-UtilAsScriptBlock(){
```

Comment:
I will remove the function for now, it doesn't belong to this PR(same as
the other remaining comments)
But what was the reason I added it?

For the BlockedFileTypes change(the one I had started), the idea is to
include a script block in the Invoke-xSharePointCommand so we can have
custom functions there and improve the code as all the blocks we have at
the moment are pure raw.

So, the change would be to change line #52 to

$invokeArgs = @{
        ScriptBlock = [ScriptBlock]::Create("if (`$null -eq
(Get-PSSnapin -Name Microsoft.SharePoint.PowerShell -ErrorAction
SilentlyContinue)) {Add-PSSnapin Microsoft.SharePoint.PowerShell}; "
+ (Get-UtilAsScriptBlock).ToString() + $ScriptBlock.ToString())

    }


The function I am wanting to add and leverage at the moment is similar to
this:
http://www.indented.co.uk/2009/08/27/merge-object/

Idea is(let's use the outgoing email settings as example).
you define this

xSPOutgoingEmailSettings  OutgoingEmail{
WebAppUrl="http://spfarm-sp:8080"
SMTPServer="smtp2.contoso.com"
ReplyToAddress="to2@email.com"
CharacterSet="65001"
PsDscRunAsCredential = $creds
}


Get-TargetResource returns
{
WebAppUrl="http://spfarm-sp:8080"
SMTPServer="smtp.contoso.com"
FromAddress="from@email.com"
ReplyToAddress="to@email.com"
CharacterSet="65001"

}

MErge-Object would return

{
WebAppUrl="http://spfarm-sp:8080"
*SMTPServer="smtp2.contoso.com <http://smtp2.contoso.com>"*
FromAddress="from@email.com"
*ReplyToAddress="to2@email.com <to2@email.com>"*
CharacterSet="65001"
}

then the Set-TargetResource  would be:

$currentSettings = Get-TargetResource $PsBoundPArams
$newSettings =  ( MergeObject -Base $newSettings -Target $current)

$webApp.UpdateMailSettings($params.SMTPServer, $params.FromAddress,
$params.ReplyToAddress, $params.CharacterSet)


A lot in thie email. Does it make sense?

On Sun, Nov 15, 2015 at 11:03 PM, Brian Farnhill <notifications@github.com>
wrote:

> In Modules/xSharePoint/Modules/xSharePoint.Util/xSharePoint.Util.psm1
> <https://github.com/PowerShell/xSharePoint/pull/95#discussion_r44870901>:
>
> > @@ -26,6 +26,10 @@ function Get-xSharePointAssemblyVersion() {
> >      )
> >      return (Get-Command $PathToAssembly).FileVersionInfo.FileMajorPart
> >  }
> > +function Get-UtilAsScriptBlock(){
>
> I don't see this function used anywhere and I'm not sure what it's purpose
> is, are you able to elaborate for me?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PowerShell/xSharePoint/pull/95/files#r44870901>.
>